src/main/java/io/nats/client/impl/NatsJetStreamPullSubscription.java

```diff
@@ -96,22 +95,7 @@ public void pullExpiresIn(int batchSize, Duration expiresIn) {
     @Override
     public void pullExpiresIn(int batchSize, long expiresInMillis) {
         durationGtZeroRequired(expiresInMillis, "Expires In");
-        _pull(batchSize, false, Duration.ofMillis(expiresInMillis));
-    }
-
-    private void _pull(int batchSize, boolean noWait, Duration expiresIn) {
-        int batch = validateGtZero(batchSize, "Pull batch size");
-        String publishSubject = js.prependPrefix(String.format(JSAPI_CONSUMER_MSG_NEXT, stream, consumerName));
-        connection.publish(publishSubject, getSubject(), getPullJson(batch, noWait, expiresIn));
-        connection.lenientFlushBuffer();
-    }
-
-    byte[] getPullJson(int batch, boolean noWait, Duration expiresIn) {
-        StringBuilder sb = JsonUtils.beginJson();
-        JsonUtils.addField(sb, "batch", batch);
-        JsonUtils.addFldWhenTrue(sb, "no_wait", noWait);
-        JsonUtils.addFieldAsNanos(sb, "expires", expiresIn);
-        return JsonUtils.endJson(sb).toString().getBytes(StandardCharsets.US_ASCII);
```

Comment:
_pull and getPullJson are not needed because the json is done in the PullRequestOptions object and the _pull code is moved to the new api that directly takes the PullRequestOptions since the other pulls can delegate to it now